### DISCUSSION TITLE
dm-raid module not found

### DISCUSSION THREAD

**Discussion by eshikhov** [siderolabs/talos#7481] [Q&A]
Hi!

To provide PV we use https://github.com/metal-stack/csi-driver-lvm, when using storageclass csi-driver-lvm-linear everything works, but when using csi-driver-lvm-mirror it gives the error: 
```
"Failed to load existing device cache from /etc/lvm/cache/.cache
    Executing: /sbin/modprobe dm-raid
modprobe: module dm-raid not found in modules.dep
  /sbin/modprobe failed: 1
  raid1: Required device-mapper target(s) not detected in your kernel.
  Run `lvcreate --help' for more information."
```
Tried to add via KernelConfig.modules add dm-raid talos gives error:
 ```
 "user: warning: [2023-07-14T21:06:37.868957414Z]: [talos] controller failed {" component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"md-raid\x5c": module not found"}"
``` 
it is possible to load the dm-raid module ?

**Comment by frezbo** [comment]
https://github.com/siderolabs/pkgs/blob/c644633324ed1e56ab19f146c04ed3984736a88a/kernel/build/config-amd64#L2272 dm raid is not enabled on talos kernel.

Also talos disallows modules to be loaded by workloads. 

The module maybe  be enabled in the talos kernel and shipped as an extension.

**Reply by smira** [reply]
not sure if that has to be an extension given that DM_MIRROR is already built-in.

**Reply by frezbo** [reply]
let's do as module, so that it's not loaded by default
